[quant] Finishing refactor for quantization test files

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37366 [quant] Finishing refactor for quantization test files**

Summary:
- we can put both fake quant module and observer module tests in the test_workflow_module.py
- added test_quantized_functional.py
- moved tests in test_numerics.py to test_quantize.py and removed test_numerics.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21282198](https://our.internmc.facebook.com/intern/diff/D21282198)

jerryzh168 committed:
[quant] Add test_workflow_module.py

Summary:
we can put both fake quant module and observer module tests in the test_workflow_module.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
-           X=hu.tensor(shapes=hu.array_shapes(1, 5,),
-                       qparams=hu.qparams(dtypes=torch.quint8)))
-    def test_forward_per_tensor(self, device, X):
```
Some of the fake-quant tests are actually tests of the aten fake-quant op. These op tests should be ideally moved to test_quantized_op.py

jerryzh168 replied:
yeah will add a todo

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-           X=hu.tensor(shapes=hu.array_shapes(1, 5,),
-                       qparams=hu.qparams(dtypes=[torch.quint8])),
-           )
-    def test_fq_module(self, device, X):
```
This is a module level test , should be ideally in test_quantized_module.py

jerryzh168 replied:
test_fake_quant.py is renamed to test_workflow_module.py

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-        dX = _fake_quantize_per_tensor_affine_grad_reference(dout, X, fq_module.scale, fq_module.zero_point, quant_min, quant_max)
-        np.testing.assert_allclose(dX.cpu().numpy(), X.grad.cpu().detach().numpy(), rtol=tolerance, atol=tolerance)
-
-    def test_fq_serializable(self):
```
This one too

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-
-        self.assertEqual(loaded_fq_module.calculate_qparams(), fq_module.calculate_qparams())
-
-    def test_fake_quant_control(self):
```
Ditto for this one, this is also a module level test

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
-           X=hu.per_channel_tensor(shapes=hu.array_shapes(1, 5,),
-           qparams=hu.qparams(dtypes=torch.quint8)))
-    def test_forward_per_channel(self, device, X):
```
This is an op level test, should move to test_quantized_op.py

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
-           X=hu.per_channel_tensor(shapes=hu.array_shapes(1, 5,),
-           qparams=hu.qparams(dtypes=torch.quint8)))
-    def test_backward_per_channel(self, device, X):
```
Same here, this is an op level test

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-           X=hu.per_channel_tensor(shapes=hu.array_shapes(1, 5,),
-           qparams=hu.qparams(dtypes=torch.quint8)))
-    @unittest.skip("temporarily disable the test")
-    def test_numerical_consistency_per_channel(self, device, X):
```
This is also an op level test

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-           X=hu.tensor(shapes=hu.array_shapes(1, 5,),
-                       qparams=hu.qparams(dtypes=torch.quint8)))
-    @unittest.skip("temporarily disable the test")
-    def test_backward_per_tensor(self, device, X):
```
This is also an op level test

raghuramank100 commented on test/quantization/test_fake_quant.py:
```diff
@@ -1,311 +0,0 @@
-                       qparams=hu.qparams(dtypes=torch.quint8)))
-    # https://github.com/pytorch/pytorch/issues/30604
-    @unittest.skip("temporarily disable the test")
-    def test_numerical_consistency_per_tensor(self, device, X):
```
Same here

raghuramank100 reviewed (approved):
I think we should refactor these differently as some of the tests are op level tests. Thanks for refactoring the code!

jerryzh168 committed:
Update on "[quant][refactor] Add test_workflow_module.py"


Summary:
we can put both fake quant module and observer module tests in the test_workflow_module.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][refactor] Add test_workflow_module.py"


Summary:
we can put both fake quant module and observer module tests in the test_workflow_module.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Finishing refactor for quantization test files"


Summary:
- we can put both fake quant module and observer module tests in the test_workflow_module.py
- added test_quantized_functional.py
- moved tests in test_numerics.py to test_quantize.py and removed test_numerics.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Finishing refactor for quantization test files"


Summary:
- we can put both fake quant module and observer module tests in the test_workflow_module.py
- added test_quantized_functional.py
- moved tests in test_numerics.py to test_quantize.py and removed test_numerics.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Finishing refactor for quantization test files"


Summary:
- we can put both fake quant module and observer module tests in the test_workflow_module.py
- added test_quantized_functional.py
- moved tests in test_numerics.py to test_quantize.py and removed test_numerics.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Finishing refactor for quantization test files"


Summary:
- we can put both fake quant module and observer module tests in the test_workflow_module.py
- added test_quantized_functional.py
- moved tests in test_numerics.py to test_quantize.py and removed test_numerics.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Finishing refactor for quantization test files"


Summary:
- we can put both fake quant module and observer module tests in the test_workflow_module.py
- added test_quantized_functional.py
- moved tests in test_numerics.py to test_quantize.py and removed test_numerics.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21282198](https://our.internmc.facebook.com/intern/diff/D21282198)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Finishing refactor for quantization test files"


Summary:
- we can put both fake quant module and observer module tests in the test_workflow_module.py
- added test_quantized_functional.py
- moved tests in test_numerics.py to test_quantize.py and removed test_numerics.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21282198](https://our.internmc.facebook.com/intern/diff/D21282198)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant] Finishing refactor for quantization test files"


Summary:
- we can put both fake quant module and observer module tests in the test_workflow_module.py
- added test_quantized_functional.py
- moved tests in test_numerics.py to test_quantize.py and removed test_numerics.py

Test Plan:
python test/test_quantization.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D21282198](https://our.internmc.facebook.com/intern/diff/D21282198)

[ghstack-poisoned]
